doc/provisioning.md

```diff
@@ -1,19 +1,28 @@
-What We Run, and Why
-====================
-
-Expressed as compactly as possible, the Provisioner is responsible for making
-sure that non-Dead machine entities in state have agents running on live
-instances; and for making sure that Dead machines, and stray instances, are
-removed and cleaned up.
-
-However, the choice of exactly what we deploy involves some subtleties. At the
-Provisioner level, it's simple: the base and the constraints we pass to the
-Environ.StartInstance come from the machine entity. But how did they get there?
-
-Bases
------
-
-Individual charms are released for different possible target base; juju
+# Provisioning
+
+Provisioning in Juju is performed by two separate workers:
```

Comment:
It reads to me like both of these will work together to provision a new machine, it could be more clear that its either or.

Reply:
Indeed, I explicitly say that it's one of those now.